### ISSUE TITLE
Clicking calendar item in calendar in CP doesn't work

### ISSUE THREAD

**Issue by boboldehampsink** [solspace/craft3-calendar#217]
Gives me a 500 Internal Server error in the console with this message:

```
TypeError: craft\web\Controller::asErrorJson(): Argument #1 ($error) must be of type string, array given, called in /app/user/vendor/solspace/craft-calendar/packages/plugin/src/Controllers/EventsApiController.php on line 302 and defined in /app/user/vendor/craftcms/cms/src/web/Controller.php:473
```

Craft 3

**Comment by kjmartens** [comment]
Hi @boboldehampsink,

What do you mean by "clicking calendar item"? Do you mean an event? And if so, where in the control panel is this happening? And which exact version of Calendar are you using? 🙂

**Comment by boboldehampsink** [comment]
Yeah an event in the month calendar view. I'm using 3.3.20

**Comment by kjmartens** [comment]
I have a few questions:

1. Are you doing anything custom on this page (including something else via module etc)?
2. Which version of PHP are you using?
3. Does this happen when clicking any event in the Month CP view? Or just some?
4. Is this something that just started recently (but worked fine before)? And if so, which version did it work fine on before?

Thanks 🙂

**Comment by boboldehampsink** [comment]
@kjmartens found the problem for ya: https://github.com/solspace/craft-calendar/blob/v4/packages/plugin/src/Controllers/EventsApiController.php#L300-L303

This should probably be "asJson" instead of "asErrorJson", which takes a string

**Comment by kjmartens** [comment]
Thanks @boboldehampsink. My developer was eyeballing that specific thing as well. We'll get it fixed shortly. 🙂

**Comment by kjmartens** [comment]
This should now be fixed in Calendar 3.3.21+ and 4.0.8+. 🙂
